yellowbrick/regressor/influence.py

```diff
@@ -0,0 +1,280 @@
+
+    outlier_percentage_ : float
+        The percentage of instances whose Cook's distance is greater than the influnce
+        threshold, the percentage is 0.0 <= p  <= 100.0.
+
+    Notes
+    -----
+    Cook's Distance is very similar to DFFITS, another diagnostic that is meant to show
+    how influential a point is in a statistical regression. Although the computed values
+    of Cook's and DFFITS are different, they are conceptually identical and there even
+    exists a closed-form formula to convert one value to another. Because of this, we
+    have chosen to implement Cook's distance rather than or in addition to DFFITS.
+    """
+
+    def __init__(
+        self, ax=None, draw_threshold=True, linefmt="C0-", markerfmt=",", **kwargs
+    ):
+        # Initializ the visualizer
+        super(CooksDistance, self).__init__(ax=ax, **kwargs)
+
+        # Set "hyperparameters"
+        self.set_params(
+            draw_threshold=draw_threshold, linefmt=linefmt, markerfmt=markerfmt,
+        )
+
+        # An internal LinearRegression used to compute the residuals and MSE
+        # This implementation doesn't support any regressor, it is OLS-specific
+        self._model = LinearRegression()
+
+    def fit(self, X, y):
+        """
+        Computes the leverage of X and uses the residuals of a
+        ``sklearn.linear_model.LinearRegression`` to compute the Cook's Distance of each
+        observation in X, their p-values and the number of outliers defined by the
+        number of observations supplied.
+
+        Parameters
+        ----------
+        X : array-like, 2D
+            The exogenous design matrix, e.g. training data.
+
+        y : array-like, 1D
+            The endogenous response variable, e.g. target data.
+
+        Returns
+        -------
+        self : CooksDistance
+            Fit returns the visualizer instance.
+        """
+        # Fit a linear model to X and y to compute MSE
+        self._model.fit(X, y)
+
+        # Leverage is computed as the diagonal of the projection matrix of X
+        # TODO: whiten X before computing leverage
+        leverage = (X * np.linalg.pinv(X).T).sum(1)
+
+        # Compute the rank and the degrees of freedom of the OLS model
+        rank = np.linalg.matrix_rank(X)
+        df = X.shape[0] - rank
+
+        # Compute the MSE from the residuals
+        residuals = y - self._model.predict(X)
+        mse = np.dot(residuals, residuals) / df
+
+        # Compute Cook's distance
+        residuals_studentized = residuals / np.sqrt(mse) / np.sqrt(1 - leverage)
+        self.distance_ = residuals_studentized ** 2 / X.shape[1]
+        self.distance_ *= leverage / (1 - leverage)
+
+        # Compute the p-values of Cook's Distance
+        # TODO: honestly this was done because it was only in the statsmodels
+        # implementation... I have no idea what this is or why its important.
+        self.p_values_ = sp.stats.f.sf(self.distance_, X.shape[1], df)
```

Comment:
honestly, this was done because it was only in the statsmodels implementation... I have no idea what this is or why it's important.